### PR TITLE
event bus

### PR DESCRIPTION
**Pull request by sergigp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,8 @@
 Quasar is a library to do [CQRS](https://martinfowler.com/bliki/CQRS.html) in Scala.
 <br> <br>
 
+**TBD: Currently this doc is deprecated. See the code to see how it works.**
```
Comment (rgomezcasas) [inline README.md:12]: 😂😂😂😂😂😂

File: src/test/scala/com/sergigp/quasar/test/AsyncEventBusTest.scala [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+class AsyncEventBusTest extends TestCase {
+  "a event bus" should {
+    "subscribe two times the same event handler without throwing error" in {
+      val eventBus = new AsyncEventBus(logger)
+      eventBus.subscribe[UserCreatedDomainEvent](EventHandlers.dummyHandler)
+      eventBus.subscribe[UserCreatedDomainEvent](EventHandlers.dummyHandler)
+    }
+
+    "throw an error if don't find a command handler for a command" in {
```
Comment (jorgeavilacardenosa) [inline src/test/scala/com/sergigp/quasar/test/AsyncEventBusTest.scala:25]: "throw an error if don't find an event handler for an event"

File: src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  private var handlers = Map.empty[Class[_], Event => Future[Any]]
+
+  override def publish[E <: Event](command: E): Future[E#EventReturnType] =
+    handlers
+      .get(command.getClass) match {
+      case Some(handler) => handler(command).map(_.asInstanceOf[E#EventReturnType])
+      case None          => Future.failed(EventHandlerNotFound(command.getClass.getSimpleName))
+    }
+
+  override def subscribe[E <: Event: ClassTag](handler: E => Future[E#EventReturnType]): Unit = {
```
Comment (rgomezcasas) [inline src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala:19]: Instead of a `subscribe` method we could pass it via constructor 🤔

**Review by rgomezcasas — APPROVED** [review]

File: src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala (hunk 3 above)
Comment (sergigp) [reply src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala:19]: circular dependency if you want to chain events

**Review by jorgeavilacardenosa — APPROVED** [review]

**Pull request merged**